assembl/fabfile.py

```diff
@@ -507,19 +551,27 @@ def migrate_local_ini(backup=False):
             os.unlink(local_file_name)
 
 
-@task
-def supervisor_restart():
-    "Restart supervisor itself."
+def supervisor_shutdown(retries=10):
+    """
+    Shutdown Assembl's supervisor
+    """
     with hide('running', 'stdout'):
         venvcmd("supervisorctl shutdown")
-    while True:
-        sleep(5)
+    print(yellow("Waiting for supervisor to shutdown"))
+    while retries > 0:
+        sleep(6)
         result = venvcmd("supervisorctl status", warn_only=True)
         if not result.failed:
             break
-        # otherwise still in shutdown mode
+        retries -= 1
     # Another supervisor, upstart, etc may be watching it, give it more time
-    sleep(5)
+    sleep(6)
+
+
+@task
+def supervisor_restart():
+    """Restart supervisor itself."""
+    execute(supervisor_shutdown)
```

Comment:
We tested `supervisorctl status` before to only run `supervisord` if it was not up. But now we shudown supervisor, so for me this "no such file" check is not needed anymore.